### ISSUE TITLE
P2D PGraphics object not rendering on Raspberry Pi ARM64

### ISSUE THREAD

**Issue by luca-panjer** [benfry/processing4#555]
## Description
I have made a class that draws a specific circle into a PGraphics object once, stores it in memory and uses the image() function to draw it to the screen when needed. This works perfectly fine on windows, but when exported to raspberry pi (64 bit), it does not show said circle on the screen. I am using a raspberry pi 4 with a fresh image of the latest 64 bit raspbian. The sketch is exported on processing windows using the built in export function.

## Expected Behavior
![image](https://user-images.githubusercontent.com/10108352/191010068-d24d92c7-9455-4675-8c04-3878750b79dc.png)
This is on windows. The left circle is drawn from a PGraphics, the right circle is drawn using the code that normally renders into the PGraphics object

## Current Behavior
![image](https://user-images.githubusercontent.com/10108352/191017141-2a12c58e-b0e5-436a-b338-65dd89eaedd9.png)
As you can see, on linux only the right cirlce is drawn. The left one, the PGraphics one, is not drawn.
There are no errors or anything in the console.

## Steps to Reproduce
[This is my sketch](https://gist.github.com/TGlev/fc4b1cb1623e5ed56c62a280c3c2243a)

## Your Environment
Raspberry pi 4, connected over HDMi to a monitor
Running the latest 64 bit raspbian with java 11.0.15.1

**Comment by luca-panjer** [comment]
I have seen this problem before when I was running on a raspberry pi 3 running a 32bit raspbian. I was running raspbian lite with an X-server instance I added to display the processing app. Moving from a raspbian lite to raspbian full image fixed this problem.

This 64bit image I am using now is based on full raspbian.

**Comment by benfry** [comment]
Processing uses Java 17, have you tried with that?

**Comment by luca-panjer** [comment]
I just upgraded to java 17 and tried again:
![image](https://user-images.githubusercontent.com/10108352/192472785-7438823a-cd94-4505-944e-8074e327ccc8.png)

Still looks the same, the PGraphics is not rendering.

**Comment by luca-panjer** [comment]
I managed to get some more logging information from the processing application. It prints this to the console:
![image](https://user-images.githubusercontent.com/10108352/194517001-b14587c6-5d76-4337-b454-7569e68a507d.png)

This happens with both P2D and P3D renderer, the default renderer does render the pgraphics just fine.

**Comment by benfry** [comment]
You may simply be running out of video memory to render the second image.

**Comment by luca-panjer** [comment]
We tried to increase the video memory, sadly still same result.
Could it be because raspberry pi has such an old OpenGL (Version 2)?

**Comment by benfry** [comment]
Yeah, could be lots of things… I know things get trickier with offscreen buffers like this on the Pi, but I don't know offhand if it's something 100% broken/unavailable, or if perhaps it's a regression in the OpenGL library if it wasn't tested on the Pi, or 64-bit RPi OS just isn't ready for prime-time, or it's something that might be fixed by tweaking some OS settings to change the framebuffer type or other video settings. Error 1282 is “invalid operation” and 1286 is “invalid framebuffer operation” so they're a sort of catch-all for “having trouble getting a lot of pixels to the screen.”

Perhaps someone with more pi-hacking experience can take a look…

**Comment by luca-panjer** [comment]
We tried running it on a 32bit raspbian as well, still same problem persists.
At this point Im thinking its the Pi4, since it did work on Pi3.

**Comment by benfry** [comment]
Ok, in that case I'd be looking for system settings for how the frame buffer is set up and making sure that's compatible. Pi 4 has a lot more power and should be much better equipped; I wonder if it's just dealing with an incompatible frame buffer.
